src/libs/actions/Policy.js

```diff
@@ -952,7 +952,7 @@ function createWorkspace(ownerEmail = '', makeMeAdmin = false, policyName = '',
         ],
     });
 
-    Navigation.isNavigationReady()
+    return Navigation.isNavigationReady()
```

Comment:
It's not good to return a promise from an action method (our docs state this and it can lead to bad code chaining if someone else sees the pattern and copies it).

I was thinking of how we could stop doing this, and here is what I think:
1. Create a new method in `App.js` called `CreateWorkspaceAndNavigateToIt()`
2. Generate the optimistic policyID in that new method and pass it to `createWorkspace()` as a parameter
3. Move this navigation logic into that new method and only have the API call here

It feels to me like that would clean up a lot of this awkwardness that we're trying to work around and it keeps all the navigation stuff in `App.js` where it should probably belong (or I could even see using a `PolicyCreationUtils.js` file at some point)

Reply:
> Eh, There are two other places that also need navigation logic.

Well, I was thinking those would just be updated to call `CreateWorkspaceAndNavigateToIt()`. Doesn't that still work like it's supposed to?

I think your alternate approach works too, but I still don't really like seeing the navigation happen inside of `createWorkspace()`. IMO, the only thing that method should be doing is creating a workspace. It should not be doing any navigation, because that is business logic that it is not concerned with.